oe_authentication.module

```diff
@@ -6,3 +6,20 @@
  */
 
 declare(strict_types = 1);
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_form__FORM_ID_alter().
+ *
+ * Remove the option to delete users while keeping the option to block them.
+ */
+function oe_authentication_form_user_cancel_form_alter(&$form, FormStateInterface $form_state) {
+  $restricted_options = [
```

Comment:
Let's introduced a new permission and only apply these restrictions if the user does not have this permission.